DataFormats/Provenance/src/ProductRegistry.cc

```diff
@@ -54,13 +52,10 @@ namespace edm {
     anyProductProduced_ = false;
 
     // propagate_const<T> has no reset() function
-    eventProductLookup_ = std::make_unique<ProductResolverIndexHelper>();
-    lumiProductLookup_ = std::make_unique<ProductResolverIndexHelper>();
-    runProductLookup_ = std::make_unique<ProductResolverIndexHelper>();
-
-    eventNextIndexValue_ = 0;
-    lumiNextIndexValue_ = 0;
-    runNextIndexValue_ = 0;
+    for (int i = InEvent; i < NumBranchTypes; ++i) {
```

Comment:
````
for(auto &p : productLookups_) { p = std::make_unique<ProductResolverIndexHelper>(); }
```
would be better since you have made the tacit assumption that `InEvent==0` but the point of the symbols is to not have to know that.

Reply:
Done